File: php/features.php
<?php
declare(strict_types=1);

// ProfanityChecker SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ProfanityCheckerFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ProfanityCheckerBaseFeature();
            case "test":
                return new ProfanityCheckerTestFeature();
            default:
                return new ProfanityCheckerBaseFeature();
        }
    }
}

File: py/features.py
# ProfanityChecker SDK feature factory

from feature.base_feature import ProfanityCheckerBaseFeature
from feature.test_feature import ProfanityCheckerTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ProfanityCheckerBaseFeature(),
        "test": lambda: ProfanityCheckerTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
